custom/icds_reports/data_pull/phone_numbers.sql

```diff
@@ -0,0 +1,110 @@
+ 039bbe4a40de499ea87b9761537dd611 | Andaman & Nicobar Islands
+ 085a5fd73e9741e69b19aaac155ba132 | Manipur
+ 2af81d10b2ca4229a54bab97a5150538 | Maharashtra
+ 9051ef4f91d54a89b89c533780857e05 | Uttar Pradesh
+ e2963f2eec18488d8016c22fb0c38a3c | Sikkim
+ 35408087a908448a849dc48e336a9fc3 | Lakshadweep
+ c7985cd779924b62b9eb863cea8e63b7 | Daman & Diu
+ 9cd4fd88d9f047088a377b7e7d144830 | Rajasthan
+ 4c84a44ded2b4b00b27373070332849f | Assam
+ 77906526407b41049f1309be323614d3 | Dadra & Nagar Haveli
+ f1cd643f0df908421abd915298ba57bc | Jharkhand
+ 3518687a1a6e4b299dedfef967f29c0c | Gujarat
+ 32c6b88173cc49ea82e804292e97c3f0 | Kerala
+ e1f3a4c8b1264042893816f1ae48819c | Delhi
+ e4cb09f85b5947689ebc35d0789e0bda | Mizoram
+*/
+
+
+COPY(SELECT
+CASE
+    WHEN length(mobile_number)=10 THEN CONCAT('+91', mobile_number)
+    WHEN length(mobile_number)=12 THEN CONCAT('+', mobile_number)
+    WHEN length(mobile_number)=13 THEN mobile_number
+END AS "Mobile Number"
+FROM ccs_record_monthly
+RIGHT JOIN awc_location loc
+ON (loc.doc_id=ccs_record_monthly.awc_id AND loc.supervisor_id=ccs_record_monthly.supervisor_id)
+WHERE
+length(mobile_number) IN (10,12,13) AND
+closed=0 AND -- or use open_in_month=1 this IS already covered by valid_in_month but if this does NOT add extra cost, keep it FOR safety
+valid_in_month=1 AND
+mobile_number IS NOT NULL AND
+month='2019-11-01' AND
+state_id='f9b47ea2ee2d8a02acddeeb491d3e175'
+) TO '/tmp/phone_numbers_mothers_state_name.csv' DELIMITER ',' CSV HEADER ENCODING 'UTF-8';
+
+/*
+Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
+Task Count: 64
+Tasks Shown: One of 64
+->  Task
+     Node: host=100.71.184.232 port=6432 dbname=icds_ucr
+     ->  Gather  (cost=1000.55..107230.08 rows=1 width=32)
+           Workers Planned: 5
+           ->  Nested Loop  (cost=0.55..106229.98 rows=1 width=32)
+                 ->  Parallel Seq Scan on ccs_record_monthly_102712 ccs_record_monthly  (cost=0.00..106149.27 rows=29 width=71)
+                       Filter: ((mobile_number IS NOT NULL) AND (closed = 0) AND (valid_in_month = 1) AND (month = '2019-11-01'::date) AND (length(mobile_number) = ANY ('{10,12,13}'::integer[])))
+                 ->  Index Scan using awc_location_indx6_102840 on awc_location_102840 loc  (cost=0.55..2.77 rows=1 width=63)
+                       Index Cond: (doc_id = ccs_record_monthly.awc_id)
+                       Filter: ((state_id = 'f9b47ea2ee2d8a02acddeeb491d3e175'::text) AND (ccs_record_monthly.supervisor_id = supervisor_id))
+(13 rows)
+*/
+
+COPY(SELECT
+CASE
+    WHEN length(mother_phone_number)=10 THEN CONCAT('+91', mother_phone_number)
+    WHEN length(mother_phone_number)=12 THEN CONCAT('+', mother_phone_number)
+    WHEN length(mother_phone_number)=13 THEN mother_phone_number
+END AS "Mobile Number"
+FROM child_health_monthly
+RIGHT JOIN awc_location loc
+ON (loc.doc_id=child_health_monthly.awc_id AND loc.supervisor_id=child_health_monthly.supervisor_id)
+WHERE
```

Comment:
Since this request if to find just the unique list  of phone numbers. How are you planning to do that distinction operation between numbers you will get from ccs_record and numbers you will get from child_health monthly.

Reply:
in the python script,
get all numbers per state in a set object(unless the count of numbers is really high in which case i might need a different way).